### PR TITLE
`azurerm_container_group` - add support for `volume.x.git_repo` 

### LINKED ISSUE #7875: Support for azurerm_container_group - mounting a Git repo to a container instance.
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Trying to create an Azure Container Group/Instances using Terraform. I see that it's possible to mount a GitRepo using Azure CLI but I do not see any reference for Terraform. Please feel free to advise, if I missed something.

### New or Affected Resource(s)

* azurerm_container_group

### Potential Terraform Configuration

resource "azurerm_container_group" "example" {
  name                = "${var.prefix}-packer"
  location            = "${azurerm_resource_group.example.location}"
  resource_group_name = "${azurerm_resource_group.example.name}"
  ip_address_type     = "public"
  os_type             = "linux"

  container {
    name     = "packer-demo"
    image    = "hashicorp/packer:full-1.6.0"
    cpu      = "1"
    memory   = "1.5"
    port     = "80"
    protocol = "tcp"

    volume {
      name       = "packer"
      mount_path = "/code/packer"
      read_only  = false
      git_repo_url = "<url> "    <--- new
     
    }
  }

### References

https://www.terraform.io/docs/providers/azurerm/r/container_group.html#volume
https://docs.microsoft.com/en-us/azure/container-instances/container-instances-volume-gitrepo

### PR DESCRIPTION
**Pull request by njuCZ**
fix #7875

### CODE REVIEW

File: azurerm/internal/services/containers/container_group_resource.go [hunk 1]
```diff
@@ -942,12 +970,17 @@ func expandContainerVolumes(input interface{}) (*[]containerinstance.VolumeMount
 
 		cv := containerinstance.Volume{
 			Name: utils.String(name),
-			AzureFile: &containerinstance.AzureFileVolume{
+		}
+
+		if gitRepoVolume := expandGitRepoVolume(volumeConfig["git_repo"].([]interface{})); gitRepoVolume != nil {
```
Comment (mbfrahry) [inline azurerm/internal/services/containers/container_group_resource.go:975]: We'll need to do some logic to make sure that either `git_repo` is specified or the share name/storage account information is specified. We'll also want to check that both aren't specified. The last check will be that all the share name and storage account information is specified if `git_repo` isn't specified. If one of those checks fail, we'll need to raise that error for the user.

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @njuCZ. This is nearly there, we just need to add some additional checks to ensure we create the volume correctly

**Comment by njuCZ** [comment]
@mbfrahry I have added some validation, and all acctest could pass, Could you have a look again when free?

**Review by mbfrahry — APPROVED** [review]
LGTM!

**Pull request merged**